components/DuplicateWarningPanel/DuplicateWarningPanel.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from 'react';
+import cx from 'classnames';
+import s from './DuplicateWarningPanel.module.scss';
+import { LegacyResident } from 'types';
+import axios from 'axios';
+import ResidentsTable from 'components/Search/results/ResidentsTable';
+
+interface Props {
+  newResident: Record<string, unknown>;
+}
+
+const DuplicateWarningPanel = ({
+  newResident,
+}: Props): React.ReactElement | null => {
+  const [matchingResidents, setMatchingResidents] = useState<LegacyResident[]>(
+    []
+  );
+
+  useEffect(() => {
+    axios
+      .get(`/api/residents`, {
+        headers: { 'x-api-key': process.env.AWS_KEY },
+        params: {
+          first_name: newResident.firstName,
+          last_name: newResident.lastName,
+        },
+      })
+      .then(({ data }) => setMatchingResidents(data.residents));
+  }, [newResident]);
+
+  console.log(matchingResidents);
+
+  if (matchingResidents?.length > 0)
```

Comment:
A nit pick, which is mostly personal preference, but I do think it makes for more logical component flow. I tend to follow the reverse pattern for these early returns. So something like this:

```tsx
if(!matchingResidents || matchingResidents.length === 0) {
  return null;
}

return <div ... /> // And the rest of your UI here...
```

I find it easier to reason about, and to add extra error cases in the future, but as I said, mostly personal preference.